src/lib/pipelines/variableGroup.ts

```diff
@@ -284,3 +284,25 @@ export const deleteVariableGroup = async (
   }
   return false;
 };
+
+/**
+ * Checks if a variable group exists
+ *
+ * @param opts optionally override spk config with Azure DevOps access options
+ * @param name Name of the variable group
+ * @returns true if the variable group exists
+ */
+export const hasVariableGroup = async (
+  opts: AzureDevOpsOpts,
+  name: string
+): Promise<boolean> => {
+  const taskClient = await getTaskAgentApi(opts);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const project = opts.project!;
+
+  const groups = await taskClient.getVariableGroups(project, name);
+  if (groups && groups.length > 0 && groups[0].name) {
+    return groups[0].name === name;
+  }
+  return false;
```

Comment:
i would suggest the you do this
```
try {
  const taskClient = await getTaskAgentApi(opts);
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const project = opts.project!;

  const groups = await taskClient.getVariableGroups(project, name);
  if (groups && groups.length > 0 && groups[0].name) {
    return groups[0].name === name;
  }
  return false;
} catch (e) {
  throw buildError(
    errorStatusCode.AZURE_VARIABLE_GROUP_ERR,
    {
      errorKey: "var-group-has-var-group-err",
        values: [project, name],
    },
    err
  );
}
```


and add to i18n.ts

"var-group-has-var-group-err": "Could not get variable group, {1} information in project, {0}.",



and since we are duplicating code in `deleteVariableGroup` function

```
export const deleteVariableGroup = async (
  opts: AzureDevOpsOpts,
  name: string
): Promise<boolean> => {
  const taskClient = await getTaskAgentApi(opts);
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const project = opts.project!;

  const groups = await taskClient.getVariableGroups(project, name);
  if (groups && groups.length > 0 && groups[0].id) {
    await taskClient.deleteVariableGroup(project, groups[0].id);
    return true;
  }
  return false;
};
```

please change `deleteVariableGroup` function
```
export const deleteVariableGroup = async (
  opts: AzureDevOpsOpts,
  name: string
): Promise<boolean> => {
  try {
    const taskClient = await getTaskAgentApi(opts);
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion

    if (hasVariableGroup(opts, name)) {
      await taskClient.deleteVariableGroup(project, groups[0].id);
      return true;
    }
    return false;
  } catch (e) {
    throw buildError(
      errorStatusCode.AZURE_VARIABLE_GROUP_ERR,
      {
        errorKey: "var-group-delete-var-group-err",
        values: [opts.project, name],
      },
      err
    );
  }
};
```

and add to i18n.ts

"var-group-delete-var-group-err": "Could not delete variable group, {1} information in project, {0}.",